storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -6839,14 +6839,8 @@ bool xb_init() {
   const char *mixed_options[4] = {NULL, NULL, NULL, NULL};
   int n_mixed_options;
 
-  if (opt_no_lock || opt_no_backup_locks) opt_lock_ddl = false;
-
-  /* sanity checks */
-  if (opt_lock_ddl && opt_lock_ddl_per_table) {
-    msg("Error: %s and %s are mutually exclusive\n", "--lock-ddl",
-        "--lock-ddl-per-table");
-    return (false);
-  }
+  if (opt_no_lock || opt_no_backup_locks || opt_lock_ddl_per_table)
+    opt_lock_ddl = false;
```

Comment:
What if user provides --lock-ddl and --lock-ddl_per_table at command line? :)

Reply:
Also since none of the tests failed, I assume there is no testcase. Please add one just to verify the fix works.